src/libtiled/mapwriter.cpp

```diff
@@ -669,6 +669,12 @@ void MapWriterPrivate::writeLayerAttributes(QXmlStreamWriter &w,
                          colorToString(layer.tintColor()));
     }
 
+    if (layer.layerType() == Layer::ImageLayerType) {
+        auto &imageLayer = static_cast<const ImageLayer&>(layer);
+        w.writeAttribute(QStringLiteral("hrepeat"), QString::number(imageLayer.repeatHorizontally()));
+        w.writeAttribute(QStringLiteral("vrepeat"), QString::number(imageLayer.repeatVertically()));
+    }
```

Comment:
Let's just put that in `writeImageLayer`. I think the only reason I did this strange handling of some tile layer attributes was because I cared about its `width` and `height` to stay together with the `x` and `y` attributes. :-)

Reply:
I wanted to do that originally, but it seemed to me that `writeImageLayer` only writes attributes to the image tag. I can have a look again later.

Reply:
That would mean you have placed the code after the `w.writeStartElement(QStringLiteral("image"));` line. You should place it right after the `writeLayerAttributes(w, imageLayer);` call.

Reply:
Done.